interfaces/builtin/serial_port.go

```diff
@@ -43,7 +43,13 @@ func (iface *SerialPortInterface) String() string {
 
 // Pattern to match allowed serial device nodes, path attributes will be
 // compared to this for validity when not using udev identification
-var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty[A-Z]{1,3}[0-9]{1,3}$")
+// Known device node patterns we need to support
+//  - ttyUSBX  (UART over USB devices)
+//  - ttyACMX  (ACM modem devices )
+//  - ttyXRUSBx  (Exar Corp. USB UART devices)
+//  - ttySX (UART serial ports)
+//  - ttyOX (UART serial ports on ARM)
+var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty[A-Z]+[0-9]+$")
```

Comment:
Given we know what we're trying to support, any reason to not be more precise:

`"^/dev/tty(USB|ACM|XRUSB|S|O)[0-9]+$"`

?